mkosi/distributions/postmarketos.py

```diff
@@ -0,0 +1,121 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+from collections.abc import Iterable
+from pathlib import Path
+
+from mkosi.config import Architecture, Config
+from mkosi.context import Context
+from mkosi.distributions import Distribution, DistributionInstaller, PackageType
+from mkosi.installer import PackageManager
+from mkosi.installer.apk import Apk, ApkRepository
+from mkosi.log import complete_step, die
+from mkosi.run import exists_in_sandbox
+from mkosi.tree import copy_tree
+
+
+class Installer(DistributionInstaller):
+    @classmethod
+    def pretty_name(cls) -> str:
+        return "postmarketOS"
+
+    @classmethod
+    def filesystem(cls) -> str:
+        return "ext4"
+
+    @classmethod
+    def package_type(cls) -> PackageType:
+        return PackageType.apk
+
+    @classmethod
+    def default_release(cls) -> str:
+        return "edge"
+
+    @classmethod
+    def default_tools_tree_distribution(cls) -> Distribution:
+        return Distribution.postmarketos
+
+    @classmethod
+    def package_manager(cls, config: Config) -> type[PackageManager]:
+        return Apk
+
+    @classmethod
+    def setup(cls, context: Context) -> None:
+        Apk.setup(context, list(cls.repositories(context)))
+
+        # Create merged /usr manually for now until our upstream (Alpine Linux) supports it:
+        # https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/85504
+        for dir in ["lib", "bin", "sbin"]:
+            (context.root / "usr" / dir).mkdir(parents=True, exist_ok=True)
+            (context.root / dir).symlink_to(f"usr/{dir}")
+
+        with complete_step("Setting up postmarketOS keyring"):
+            # Create keys directory in sandbox
+            keys_dir = context.sandbox_tree / "etc/apk/keys"
+            keys_dir.mkdir(parents=True, exist_ok=True)
+
+            # Copy keys from various sources (if they exist)
+            tools_usr = context.config.tools() / "usr"
```

Comment:
This is a bit too much variable indirection for me, just inline this into the 3 callsites below